pygeoweaver/sc_detail.py

```diff
@@ -1,66 +1,81 @@
 """
 Detail subcommand
 """
+import json
 
-import subprocess
-
+import pandas as pd
 import requests
 from pygeoweaver.constants import *
 
 from pygeoweaver.utils import (
     download_geoweaver_jar,
-    get_geoweaver_jar_path,
-    get_java_bin_path,
-    get_root_dir,
+    create_table,
 )
+import ipywidgets as widgets
+from IPython.display import display, HTML
 
 
 def detail_workflow(workflow_id):
     if not workflow_id:
         raise RuntimeError("Workflow id is missing")
     download_geoweaver_jar()
-    subprocess.run(
-        [
-            get_java_bin_path(),
-            "-jar",
-            get_geoweaver_jar_path(),
-            "detail",
-            f"--workflow-id={workflow_id}",
-        ],
-        cwd=f"{get_root_dir()}/",
-    )
+    url = f"{GEOWEAVER_DEFAULT_ENDPOINT_URL}/web/detail"
+    headers = {'Content-Type': 'application/x-www-form-urlencoded'}
+    form_data = {'type': 'workflow', 'id': workflow_id}
+    d = requests.post(url=url, data=form_data, headers=headers)
+    d = d.json()
+    d['nodes'] = json.loads(d['nodes'])
+    try:
+        from IPython import get_ipython
```

Comment:
this section of code is repeated in all the places. Can you make it a function and reuse it when being used?